tests/unit-tests/dds/DCPS/PrinterValueWriter.cpp

```diff
@@ -0,0 +1,447 @@
+  pvw.write_int16(6);
+  pvw.end_struct_member();
+  pvw.end_struct();
+  EXPECT_STREQ(pvw.str().c_str(), "    aField: 5\n    bField: 6");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, begin_array)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_array();
+  EXPECT_STREQ(pvw.str().c_str(), "");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, end_array)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_array();
+  pvw.end_array();
+  EXPECT_STREQ(pvw.str().c_str(), "");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, begin_sequence)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_sequence();
+  EXPECT_STREQ(pvw.str().c_str(), "");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, end_sequence)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_sequence();
+  pvw.end_sequence();
+  EXPECT_STREQ(pvw.str().c_str(), "");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, begin_element)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_sequence();
+  pvw.begin_element(0);
+  EXPECT_STREQ(pvw.str().c_str(), "    [0]: ");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, end_element)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_sequence();
+  pvw.begin_element(0);
+  pvw.write_int16(5);
+  pvw.end_element();
+  EXPECT_STREQ(pvw.str().c_str(), "    [0]: 5");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, complete_sequence)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_sequence();
+  pvw.begin_element(0);
+  pvw.write_int16(5);
+  pvw.end_element();
+  pvw.begin_element(1);
+  pvw.write_int16(6);
+  pvw.end_element();
+  pvw.end_sequence();
+  EXPECT_STREQ(pvw.str().c_str(), "    [0]: 5\n    [1]: 6");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, complete_sequence_write_array)
+{
+  const ACE_CDR::Short i[]= {5, 6};
+  PrinterValueWriter pvw;
+  pvw.begin_sequence();
+  pvw.write_int16_array(&i[0], 2);
+  pvw.end_sequence();
+  EXPECT_STREQ(pvw.str().c_str(), "    [0]: 5\n    [1]: 6");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, complete_array)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_array();
+  pvw.begin_element(0);
+  pvw.write_int16(5);
+  pvw.end_element();
+  pvw.begin_element(1);
+  pvw.write_int16(6);
+  pvw.end_element();
+  pvw.end_array();
+  EXPECT_STREQ(pvw.str().c_str(), "    [0]: 5\n    [1]: 6");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, complete_array_write_array)
+{
+  ACE_CDR::Short const i[2] = {5, 6};
+  PrinterValueWriter pvw;
+  pvw.begin_array();
+  pvw.write_int16_array(&i[0], 2);
+  pvw.end_array();
+  EXPECT_STREQ(pvw.str().c_str(), "    [0]: 5\n    [1]: 6");
+}
+
+TEST(dds_DCPS_PrinterValueWriter, complete_struct_with_complete_array)
+{
+  PrinterValueWriter pvw;
+  pvw.begin_struct();
+  pvw.begin_struct_member(OpenDDS::XTypes::MemberDescriptor("a", false));
+  pvw.begin_array();
+  pvw.begin_element(0);
+  pvw.write_int16(5);
+  pvw.end_element();
+  pvw.begin_element(1);
+  pvw.write_int16(6);
+  pvw.end_element();
+  pvw.end_array();
+  pvw.end_struct_member();
+  pvw.end_struct();
+  EXPECT_STREQ(pvw.str().c_str(), "    a: \n        [0]: 5\n        [1]: 6");
```

Comment:
If you don't want to change this to avoid the trailing whitespace, this should make the lint script stop complaining about it.

```suggestion
  pvw.end_struct();
  // lint.pl ignores whitespace_before_newline_in_string on next line
  EXPECT_STREQ(pvw.str().c_str(), "    a: \n        [0]: 5\n        [1]: 6");
```